### PR TITLE
rustcat: initial commit

### PR DESCRIPTION
**Pull request by robiot**

### CODE REVIEW

**Comment by D4imyo** [comment]
hey, please look at `blackarch-pkgbuilds` repo so you can compare your work with it, like the order of variables,..etc

File: packages/rustcat/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    'git'
+)
+md5sums=('SKIP')
+
+source=("$pkgname::git+https://github.com/robiot/rustcat.git")
+
+pkgver() {
+  cd $pkgname
+
+  echo $(git rev-list --count HEAD).$(git rev-parse --short HEAD)
```
Comment (D4imyo) [inline packages/rustcat/PKGBUILD:24]: https://github.com/BlackArch/blackarch-pkgbuilds/blob/057a20ced2a87770fb83d169c56956fab9055ccb/PKGBUILD-generic-git#L20-L24

**Review by D4imyo — CHANGES_REQUESTED** [review]

**Comment by robiot** [comment]
Thanks @D4imyo, I will fix this as fast as I can :)

File: packages/rustcat/PKGBUILD [hunk 2]
```diff
@@ -3,34 +3,35 @@
...
-md5sums=('SKIP')
+sha512sums=('SKIP')
 
 source=("$pkgname::git+https://github.com/robiot/rustcat.git")
 
 pkgver() {
   cd $pkgname
 
-  echo $(git rev-list --count HEAD).$(git rev-parse --short HEAD)
+  echo $(git describe --tags --abbrev=0 | cut -c 2-)
```
Comment (noraj) [inline packages/rustcat/PKGBUILD:24]: Use `git describe --long --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g'`, please follow https://github.com/BlackArch/blackarch-pkgbuilds

**Review by noraj — CHANGES_REQUESTED** [review]

File: packages/rustcat/PKGBUILD (hunk 2 above)
Comment (robiot) [reply packages/rustcat/PKGBUILD]: Fixed it!

File: packages/rustcat/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+_binname=rc
+pkgver=v1.1.2.r0.g2e7879c
+pkgrel=1
+groups=("blackarch", "blackarch-networking")
+pkgdesc="Rustcat - A Modern Port Listener & Reverse Shell"
+url="https://github.com/robiot/rustcat"
+license=('MIT')
+arch=('x86_64')
+makedepends=(
+    'rustup'
```
Comment (noraj) [inline packages/rustcat/PKGBUILD:14]: put them on one line until it is 80 chars line length

File: packages/rustcat/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  cd $pkgname
+
+  echo $(git describe --long --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g')
+}
+
+build() {
+    cd $pkgname
+
+    # Build tag version
+    git checkout $(git describe --tags --abbrev=0) &> /dev/null
```
Comment (noraj) [inline packages/rustcat/PKGBUILD:31]: keep last commit, remove the checkout

**Review by noraj — CHANGES_REQUESTED** [review]

**Comment by noraj** [comment]
Also testing it in the current state fails:

```
$ ba-dev -b
...
error: no override and no default toolchain set
==> ERROR: A failure occurred in build().
    Aborting...
==> ERROR: Build failed, check /home/noraj/blackarch_chroot/blackarch/build
Failed to build the package.
```

**Comment by robiot** [comment]
> Also testing it in the current state fails:
> 
> ```
> $ ba-dev -b
> ...
> error: no override and no default toolchain set
> ==> ERROR: A failure occurred in build().
>     Aborting...
> ==> ERROR: Build failed, check /home/noraj/blackarch_chroot/blackarch/build
> Failed to build the package.
> ```

Fixed this, and the other errors. Also tried it with ba-dev and it succeded

**Comment by noraj** [comment]
Builds correctly: `ba-dev -e 'rc -h' -p rustcat-v1.1.2.r2.g2cae2f9-1-x86_64.pkg.tar.zst`

**Review by noraj — APPROVED** [review]

**Pull request merged**